server/middleware.js

```diff
@@ -0,0 +1,72 @@
+const path = require("path");
+
+/**
+ * This middleware functions allows us to redirect any api request to a file like bundle.js back to the original /bundle.js.
+ * When we don't redirect these files back to their base route,
+ * server side errors are thrown due to the server trying to find the /Puzzle/Selection/bundle.js route, etc. upon page refresh.
+ * @param {Express.Request} req - an ExpressJS variable holding request data from an API call
+ * @param {Express.Response} res - an ExpressJS variable holding response data for an API call.
+ * @param {Function} next - the next function to execute.
+ * @returns - a server redirect if the route is for a redirectFile
+ */
+const redirectBundleManifest = (req, res, next) => {
+  const redirectFileNames = [
+    "bundle.js",
+    "bundle.css",
+    "manifest.json",
+    "favicon.ico",
+  ];
+
+  redirectFileNames.forEach((fileName) => {
+    if (req.path !== `/${fileName}` && req.path.includes(fileName)) {
+      return res.redirect(`/${fileName}`);
+    }
+  });
+
+  next();
+};
+
+/**
+ * This function checks whether or not {req/path} is an excepted path.
+ * - If it is, the function moves onto the next middleware.
+ * - If it is not, then it checks if the request has a sessionID, which corresponds to the user having logged in.
+ *   - If no sessionID is found on the request, then the user has not logged in, and the middleware returns a 401 status.
+ * @param {Express.Request} req - an ExpressJS variable holding request data from an API call
+ * @param {Express.Response} res - an ExpressJS variable holding response data for an API call.
+ * @param {Function} next - the next function to execute.
+ * @returns `res` with a 401 status along with the `index.html` file if no session token was found in `req` and the route requires one.
+ */
+const logRouteAndCheckAuthorization = (req, res, next) => {
+  const exceptedRoutes = [
+    "/api/login",
+    "/api/signup",
+    "/",
+    "/bundle.css",
+    "/bundle.js",
+    "/manifest.json",
+    "/logo192.png",
```

Comment:
Same thing here, logo192.png doesn't exist. I have seen some messages in my console about it trying to find logo192.png so I wonder if this is why it's trying to look for it when it no longer exists

Reply:
That might be due to the `manifest.json`, but not 100% sure on that. Either way, I can definitely remove that